tests/integrationv2/test_sslyze.py

```diff
@@ -0,0 +1,251 @@
+import pytest
+import sslyze
+
+from configuration import available_ports, ALL_TEST_CIPHERS, ALL_TEST_CERTS
+from common import ProviderOptions, Protocols, Cipher, Ciphers, Certificates, Curves
+from fixtures import managed_process
+from providers import S2N
+from utils import get_parameter_name, invalid_test_parameters
+
+HOST = "127.0.0.1"
+
+PROTOCOLS_TO_TEST = [
+    Protocols.SSLv3,
+    Protocols.TLS10,
+    Protocols.TLS11,
+    Protocols.TLS12,
+    Protocols.TLS13
+]
+
+SSLYZE_SCANS_TO_TEST = {
+    sslyze.ScanCommand.ROBOT,
+    sslyze.ScanCommand.SESSION_RESUMPTION,
+    sslyze.ScanCommand.TLS_COMPRESSION,
+    sslyze.ScanCommand.TLS_1_3_EARLY_DATA,
+    sslyze.ScanCommand.TLS_FALLBACK_SCSV,
+    sslyze.ScanCommand.HEARTBLEED,
+    sslyze.ScanCommand.OPENSSL_CCS_INJECTION,
+    sslyze.ScanCommand.SESSION_RENEGOTIATION
+}
+
+CERTS_TO_TEST = [
+    cert for cert in ALL_TEST_CERTS if cert.name not in {
+        "RSA_PSS_2048_SHA256"  # sslyze doesn't like this cert
+    }
+]
+
+CIPHER_SUITE_SCANS = {
+    Protocols.SSLv3.value: sslyze.ScanCommand.SSL_3_0_CIPHER_SUITES,
+    Protocols.TLS10.value: sslyze.ScanCommand.TLS_1_0_CIPHER_SUITES,
+    Protocols.TLS11.value: sslyze.ScanCommand.TLS_1_1_CIPHER_SUITES,
+    Protocols.TLS12.value: sslyze.ScanCommand.TLS_1_2_CIPHER_SUITES,
+    Protocols.TLS13.value: sslyze.ScanCommand.TLS_1_3_CIPHER_SUITES
+}
+
+
+def get_scan_attempts(scan_results):
```

Comment:
This is more of a general comment, the processing of these scan results is getting complex, and it makes it difficult to reason about how this test works. I noticed in our [previous test ](https://github.com/aws/s2n-tls/blob/main/tests/integration/s2n_sslyze_test.py#L88) we are putting the scan results into a json format and then parsing the json object, which only requires about 3 lines of code. That looks cleaner than all the parsing here. Do you think switching to a json object would make this test easier to read? Or can you think of some other way of cleaning up the parsing logic here?

Reply:
Hopefully this change (5aa01edafafd4e8a7dbed5693fe2c7025f1c0605) made things less complex and more readable. Let me know what you think, and if there's anything else you think could be made simpler.

Reply:
readability has improved :-)

Reply:
Yeah this version is much easier to review!